plugin/awesoMD/plugin.js

```diff
@@ -454,35 +467,116 @@ const Plugin = () => {
+					firstHeadingProcessingDone = true
+				} else {
+					const previousLine = lines[index - 1] || ''
+					if (previousLine !== options.slideSeparator) {
+						result.push(options.slideSeparator)
+					}
+				}
+			}
+			result.push(line)
+		})
+		markdown = result.join('\n')
+		return markdown
+	}
+
+	/**
+	 * Separates the inline metadata and content for slide having inline metadata in yaml block as
+	 *
+	 * ```yaml
+	 * metadata_key1: metadata_value1
+	 * metadata_key2: metadata_value2
+	 * ```
+	 */
+	function extractYAMLMetadata(markdown, options) {
+		const markdownParts = yamlRegex.exec(markdown)
+		yamlRegex.lastIndex = 0
+		if (markdownParts && markdownParts[2]) {
+			const metadata = markdownParts[2]
+			markdown = markdownParts[3] || ''
+
+			try {
+				const metadataYAML = yaml.load(metadata)
+				if (metadataYAML === undefined) {
+					throw new Error("The inline metadata is not valid.")
+				}
+				options.metadata = { ...options.metadata, ...metadataYAML }
+				options.attributes = 'class=' + (options.metadata.slide || '')
+			} catch (error) {
+				console.error(error)
+				markdown = error.message
+			}
+		}
+		return [markdown, options]
+	}
+
+	/**
+	 * Separates the inline metadata and content for slides having metadata as
+	 *
+	 * ::metadata_key1:metadata_value1 ::metadata_key2:metadata_value2
+	 */
+	function extractInlineMetadata(markdown, options) {
+		const inlineMetadata = {}
+		const matches = markdown.match(headingWithMetadataRegex)
+
+		if (matches && matches[0]) {
+			const metadataMatches = matches[0].match(metadataRegex)
+			if (metadataMatches) {
+				metadataMatches.forEach(metadataMatch => {
+					const [key, value] = metadataMatch.replace('::', '').split(':')
+					inlineMetadata[key.trim()] = value.trim()
+					const metadataPattern = new RegExp(`::\\b${key.trim()}\\b:\\s*${value.trim()}`)
+					markdown = markdown.replace(metadataPattern, '')
+				})
+			}
+		}
+
+		options.metadata = { ...options.metadata, ...inlineMetadata }
+		options.attributes = 'class=' + (options.metadata.slide || '')
+		return [markdown, options]
+	}
+
 	/**
 	 * Separates the inline metadata and content for each slide
 	 *
 	 * Returns updated options with the inline metadata and
 	 * updated markdown without the inline metadata for each slide
 	 */
 	function separateInlineMetadataAndMarkdown(markdown, options) {
-		const yamlRegex =  /```(yaml|yml)\n([\s\S]*?)```(\n[\s\S]*)?/g;
-		if (yamlRegex.test(markdown)){
-			yamlRegex.lastIndex = 0;
-
-			const markdownParts = yamlRegex.exec(markdown)
-			const metadata = markdownParts[2] || {}
-			markdown = markdownParts[3] || ''
-			if (metadata){
-				try {
-					const metadataYAML = yaml.load(metadata);
-					if (metadataYAML === undefined) {
-						throw new Error("The inline metadata is not valid.")
+		const yamlMetadata = yamlRegex.test(markdown)
+		const newMetadata = headingWithMetadataRegex.test(markdown)
+		yamlRegex.lastIndex = 0
+
+		if (options.separateByHeading) {
```

Comment:
please add a test that checks if the behavior is correct when this value is set to `true`